source/agora/common/Amount.d

```diff
@@ -0,0 +1,239 @@
+/*******************************************************************************
+
+    Defines a monetary type used in the blockchain
+
+   This struct can hold an amount between 0 and 500_000_000 * 10^7,
+   as there is an absolute maximum supply of 500M and each coin can be divided
+   in up to 10^7 units.
+
+   This struct does not expose operation overloading on purpose.
+   Having operator overloading would mean that any error should be reported
+   as either an `assert` being triggered or an `Exception` being thrown.
+   The convenience of using `a + b` would likely mean some places wouldn't
+   be properly bound checked, which would in turn open a DoS.
+   Instead, we provide two kind of functions for operations:
+   `bool OPNAME(ref Amount res, const Type)` and `Amount mustOPNAME(const Type)`
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.Amount;
+
+/// Ditto
+public struct Amount
+{
+    import agora.common.Deserializer;
+    import agora.common.Hash;
+    import agora.common.Serializer;
+
+    import core.checkedint;
+
+    /// Number of units ('cents') per coins
+    public static immutable Amount UnitPerCoin   = Amount(10_000_000, true);
+    /// Maximum number of BOA coins that can ever be in circulation
+    public static immutable ulong  MaxCoinSupply = 500_000_000;
+    /// Maximum amount of money that can ever be in circulation
+    public static immutable Amount MaxUnitSupply =
+        Amount(UnitPerCoin.value * MaxCoinSupply, true);
+
+    /// Internal data storage
+    private ulong value;
+
+    /// Support for hashing
+    public void computeHash(scope HashDg dg) const nothrow @nogc @safe
```

Comment:
https://github.com/bpfkorea/agora/blob/v0.x.x/doc/Style.md#functions

:P

Reply:
Make the suggestion instead you dingus